modyn/storage/src/internal/filesystem_wrapper/local_filesystem_wrapper.cpp

```diff
@@ -66,22 +66,15 @@ uint64_t LocalFilesystemWrapper::get_file_size(const std::string& path) {
   return static_cast<int64_t>(std::filesystem::file_size(path));
 }
 
-template <typename TP>
-std::time_t to_time_t(TP tp) {
-  using namespace std::chrono;
-  auto sctp = time_point_cast<system_clock::duration>(tp - TP::clock::now() + system_clock::now());
-  return system_clock::to_time_t(sctp);
-}
-
 int64_t LocalFilesystemWrapper::get_modified_time(const std::string& path) {
   ASSERT(is_valid_path(path), fmt::format("Invalid path: {}", path));
   ASSERT(exists(path), fmt::format("Path does not exist: {}", path));
   static_assert(sizeof(int64_t) >= sizeof(std::time_t), "Cannot cast time_t to int64_t");
 
-  const auto modified_time = std::filesystem::last_write_time(path);
-  const auto cftime = to_time_t(modified_time);
-  return static_cast<int64_t>(cftime);
-
+  // there is no portable way to get the modified time of a file in C++17 and earlier
+  struct stat file_attribute {};
+  stat(path.c_str(), &file_attribute);
+  return static_cast<int64_t>(file_attribute.st_mtime);
   /* C++20 version, not supported by compilers yet */
```

Comment:
this is needed as the previous C++ way is not accurate. Sometimes a different modified time is read into the system (due to inaccurate data type converstion). I rewrite it in C because prior to C++20 there is no absolutely correct way in C++.

Reply:
I actually don't understand the problem yet. What do you mean by "Sometimes a different modified time is read into the system (due to inaccurate data type converstion)"? At the time, I researched this quite a bit and this seemed to be the most reliable implementation. We should maybe also check whether compiler support for C++20 chrono has improved since writing this. 

Anyways, I don't understand yet what the problem is with the std::filesytem approach, since that should be realiable. Could you maybe clarify why you intend to change this?

Reply:
We here need to convert `std::filesystem::file_time_type` to `time_t`.
The previous way resorts to `TP::clock::now()` and `system_clock::now()`, which are two separate calls to the current time and might produce a difference. So it is inaccurate. Also @jenny011 confirmed this inaccuracy in her experiments, where the some files with modified time `**00` is read as `**99`.

I believe the previous implementation consulted this [stackoverflow QA](https://stackoverflow.com/questions/61030383/how-to-convert-stdfilesystemfile-time-type-to-time-t) because the impl is exactly the same.

Reply:
Okay, then we use the C implementation.